jerry-core/ecma/builtin-objects/ecma-builtin-promise.c

```diff
@@ -38,6 +41,616 @@
+  ecma_value_t already_called =  ecma_op_object_get (function_p, str_already_called);
+
+  JERRY_ASSERT (ecma_is_value_boolean (already_called));
+
+  /* 2. */
+  if (ecma_is_value_true (already_called))
+  {
+    ecma_fast_free_value (already_called);
+    ecma_deref_ecma_string (str_already_called);
+
+    return ret;
+  }
+
+  /* 3. */
+  ecma_op_object_put (function_p,
+                      str_already_called,
+                      ecma_make_boolean_value (true),
+                      false);
+
+  ecma_string_t *str_index = ecma_new_ecma_string_from_uint32 (ECMA_PROMISE_PROPERTY_INDEX);
+  ecma_string_t *str_value = ecma_new_ecma_string_from_uint32 (ECMA_PROMISE_PROPERTY_VALUE);
+  ecma_string_t *str_capability = ecma_new_ecma_string_from_uint32 (ECMA_PROMISE_PROPERTY_CAPABILITY);
+  ecma_string_t *str_remaining = ecma_new_ecma_string_from_uint32 (ECMA_PROMISE_PROPERTY_REMAINING_ELEMENT);
+
+  /* 4-7. */
+  ecma_value_t index_val = ecma_op_object_get (function_p, str_index);
+  ecma_value_t value_array = ecma_op_object_get (function_p, str_value);
+  ecma_value_t capability = ecma_op_object_get (function_p, str_capability);
+  ecma_value_t remaining = ecma_op_object_get (function_p, str_remaining);
+
+  JERRY_ASSERT (ecma_is_value_integer_number (index_val));
+
+  /* 8. */
+  ecma_string_t *index_str = ecma_new_ecma_string_from_uint32 ((uint32_t) ecma_get_integer_from_value (index_val));
+
+  ecma_op_object_put (ecma_get_object_from_value (value_array),
+                      index_str,
+                      argv[0],
+                      false);
+  ecma_deref_ecma_string (index_str);
+
+  /* 9-10. */
+  if (ecma_builtin_promise_remaining_inc_or_dec (remaining, false) == 0)
+  {
+    ecma_string_t *str_resolve = ecma_new_ecma_string_from_uint32 (ECMA_PROMISE_PROPERTY_RESOLVE);
+    ecma_value_t resolve = ecma_op_object_get (ecma_get_object_from_value (capability),
+                                               str_resolve);
+    ecma_deref_ecma_string (str_resolve);
+    ret = ecma_op_function_call (ecma_get_object_from_value (resolve),
+                                 ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED),
+                                 &value_array,
+                                 1);
+    ecma_free_value (resolve);
+  }
+
+  ecma_free_value (remaining);
+  ecma_free_value (capability);
+  ecma_free_value (value_array);
+  ecma_fast_free_value (index_val);
+  ecma_fast_free_value (already_called);
+  ecma_deref_ecma_string (str_already_called);
+  ecma_deref_ecma_string (str_index);
+  ecma_deref_ecma_string (str_value);
+  ecma_deref_ecma_string (str_capability);
+  ecma_deref_ecma_string (str_remaining);
+
+  return ret;
+} /* ecma_builtin_promise_all_handler */
+
+/**
+ * Runtime Semantics: PerformPromiseAll.
+ *
+ * See also:
+ *         ES2015 25.4.4.1.1
+ *
+ * @return ecma value of the new promise.
+ *         Returned value must be freed with ecma_free_value.
+ */
+inline static ecma_value_t
```

Comment:
Because those functions ( `ecma_builtin_promise_do_all`, `ecma_builtin_promise_do_race` and `ecma_builtin_promise_reject_abrupt`) are only called in one certain place. I get out those code and form a function to enhance the readability.
So I think "inline" will prevent the function-call-overhead, and at the same time, it will not increase the code size.